test/unit/TokenController/transferStakingPoolOwnership.js

```diff
@@ -0,0 +1,117 @@
+    ).to.not.be.reverted;
+
+    expect(await tokenController.getManagerStakingPools(newManager.address)).to.be.deep.equal([]);
+    expect(await tokenController.isStakingPoolManager(newManager.address)).to.be.equal(false);
+    expect(await tokenController.getStakingPoolManager(poolId)).to.be.equal(AddressZero);
+
+    expect(await tokenController.getManagerStakingPools(oldManager.address)).to.be.deep.equal([]);
+    expect(await tokenController.isStakingPoolManager(oldManager.address)).to.be.equal(false);
+  });
+
+  it('should set new address of manager of pools, and remove from old', async function () {
+    const { tokenController } = this.contracts;
+    const {
+      members: [oldManager, newManager],
+      internalContracts: [internalContract],
+    } = this.accounts;
+
+    // Set old manager
+    await tokenController.connect(internalContract).assignStakingPoolManager(poolId, oldManager.address);
+
+    await expect(
+      tokenController.connect(internalContract).transferStakingPoolsOwnership(oldManager.address, newManager.address),
+    ).to.not.be.reverted;
+
+    // Check new manager
+    const poolsNewManager = await tokenController.getManagerStakingPools(newManager.address);
+    expect(poolsNewManager).to.be.deep.equal([poolId]);
+    expect(await tokenController.isStakingPoolManager(newManager.address)).to.be.equal(true);
+    expect(await tokenController.getStakingPoolManager(poolId)).to.be.equal(newManager.address);
+
+    // Check old manager
+    const poolsOldManager = await tokenController.getManagerStakingPools(oldManager.address);
+    expect(poolsOldManager).to.be.deep.equal([]);
+    expect(await tokenController.isStakingPoolManager(oldManager.address)).to.be.equal(false);
+  });
+
+  it('should transfer 20 pools from old manager to new manager', async function () {
+    const { tokenController } = this.contracts;
+    const {
+      members: [oldManager, newManager],
+      internalContracts: [internalContract],
+    } = this.accounts;
+
+    const assignPromises = [];
+    for (let i = 0; i < 20; i++) {
+      assignPromises.push(tokenController.connect(internalContract).assignStakingPoolManager(i, oldManager.address));
+    }
+    await Promise.all(assignPromises);
+
+    await tokenController
+      .connect(internalContract)
+      .transferStakingPoolsOwnership(oldManager.address, newManager.address);
+
+    // Check new manager
+    const poolsNewManager = await tokenController.getManagerStakingPools(newManager.address);
+    expect(poolsNewManager.length).to.be.equal(20);
+    expect(await tokenController.isStakingPoolManager(newManager.address)).to.be.equal(true);
+
+    // Check old manager
+    const poolsOldManager = await tokenController.getManagerStakingPools(oldManager.address);
+    expect(poolsOldManager.length).to.be.equal(0);
+    expect(await tokenController.isStakingPoolManager(oldManager.address)).to.be.equal(false);
+  });
+
+  // TODO: not sure if we want this behavior
```

Comment:
given we only use it within switchMembership now, this is fine as the new address becomes a member.
however, switchMembership should check that the `to` address cannot be 0. I'll create a separate issue for this